src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImage.groovy

```diff
@@ -118,6 +131,17 @@ class DockerBuildImage extends AbstractDockerRemoteApiTask implements RegistryCr
         quiet.set(false)
         pull.set(false)
         cacheFrom.empty()
+        String safeTaskPath = path.replaceFirst("^:", "").replaceAll(":", "_")
+        imageIdFile.set(project.layout.buildDirectory.file(".docker/${safeTaskPath}-imageId.txt"))
+
+        outputs.upToDateWhen {
+            if(imageIdFile.get().asFile.exists()) {
+                // Get all images in docker registry and make sure the imageId is found in there
+                getDockerClient().listImagesCmd().exec().any {
```

Comment:
I just wonder if this can hit performance if registry contains large number of images. Can we use here `DockerInspectImage` task?

The second thing is would if it work with private registries?

Reply:
Agreed. At the very least it would be cheaper all things considered vs iterating over everything.

Reply:
You're right for docker inspect. On a normal repo (600 images) it takes 0.6 seconds for the list, 0.03 for the inspect when image found. If image not present, inspect takes approx the same time than list. So I suppose we should use inspect and scan the return code.

I am not sure to understand the question for private registries...

Reply:
Here is the code with the inspect. Unfortunately, I found no other way than to use the exception handling in my normal workflow. This is how inspect is implemented. Still much faster though... Let me know what you think.